iroh-bytes/src/store/file/tests.rs

```diff
@@ -0,0 +1,892 @@
+        };
+        assert_eq!(size, SIZE);
+        assert_eq!(tt.hash(), &hash);
+        assert_eq!(actual.db, Some(expected));
+        assert!(outboard.is_empty());
+    }
+    {
+        const SIZE: u64 = MID_SIZE;
+        let mid = Bytes::from(random_test_data(SIZE as usize));
+        let path = tempdir.path().join("mid.data");
+        std::fs::write(&path, &mid).unwrap();
+        let (outboard, hash) = raw_outboard(&mid);
+        let (tt, size) = db
+            .import_file(
+                path.clone(),
+                ImportMode::TryReference,
+                BlobFormat::Raw,
+                np(),
+            )
+            .await
+            .unwrap();
+        let actual = db.entry_state(*tt.hash()).await.unwrap();
+        let expected = EntryState::Complete {
+            data_location: DataLocation::External(vec![path.clone()], SIZE),
+            outboard_location: OutboardLocation::Inline(Bytes::from(outboard)),
+        };
+        assert_eq!(size, SIZE);
+        assert_eq!(tt.hash(), &hash);
+        assert_eq!(actual.db, Some(expected));
+        assert_eq!(mid, std::fs::read(path).unwrap());
+        assert!(!db.owned_data_path(&hash).exists());
+    }
+    drop(tempdir);
+}
+
+#[tokio::test]
+async fn import_file_error_cases() {
+    let np = IgnoreProgressSender::<ImportProgress>::default;
+    let (tempdir, db) = create_test_db().await;
+    // relative path is not allowed
+    {
+        let path = PathBuf::from("relativepath.data");
+        let cause = db
+            .import_file(path, ImportMode::Copy, BlobFormat::Raw, np())
+            .await
+            .unwrap_err();
+        assert_eq!(cause.kind(), io::ErrorKind::InvalidInput);
+    }
+    // file does not exist
+    {
+        let path = tempdir.path().join("pathdoesnotexist.data");
+        let cause = db
+            .import_file(path, ImportMode::Copy, BlobFormat::Raw, np())
+            .await
+            .unwrap_err();
+        assert_eq!(cause.kind(), io::ErrorKind::InvalidInput);
+    }
+    // file is a directory
+    {
+        let path = tempdir.path().join("pathisdir.data");
+        std::fs::create_dir_all(&path).unwrap();
+        let cause = db
+            .import_file(path, ImportMode::Copy, BlobFormat::Raw, np())
+            .await
+            .unwrap_err();
+        assert_eq!(cause.kind(), io::ErrorKind::InvalidInput);
+    }
+    // // file is not readable for the store
+    // #[cfg(unix)]
+    // {
+    //     let path = tempdir.path().join("forbidden.data");
+    //     let data = random_test_data(1024);
+    //     std::fs::write(&path, &data).unwrap();
+    //     std::fs::set_permissions(&path, PermissionsExt::from_mode(0o0)).unwrap();
+    //     let cause = db
+    //         .import_file(path, ImportMode::Copy, BlobFormat::Raw, np())
+    //         .await
+    //         .unwrap_err();
+    //     assert_eq!(cause.kind(), io::ErrorKind::PermissionDenied);
```

Comment:
why is this commented out?

Reply:
This was my attempt to cover some of the not so happy cases, but it turned out that using PermissionExt to set some thing readonly did not have the desired effect on windows.

Maybe I can reenable them only on unix or come up with another way to trigger an error.